### PR TITLE
Fix training lora arg checking and exiting

### PR DESCRIPTION
**Pull request by Maxusmusti**
Currently is_padding_free is broken (will always exit) due to current lora arg processing. This PR fixes that.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: src/instructlab/configuration.py [hunk 1]
```diff
@@ -1130,24 +1130,29 @@ def map_train_to_library(ctx, params):
...
-        lora = True
+        lora_options = True
         lora_args.target_modules = params["lora_target_modules"]
     if params["lora_quantize_dtype"] is not None:
-        lora = True
+        lora_options = True
         lora_args.quantize_data_type = params["lora_quantize_dtype"]
-    if lora and params["is_padding_free"]:
-        ctx.fail("Cannot combine LoRA with a padding free model.")
+    if lora_enabled and params["is_padding_free"]:
```
Comment (cdoern) [inline src/instructlab/configuration.py:1150]: so lora_enabled == lora_rank. this makes sense

File: src/instructlab/configuration.py [hunk 2]
```diff
@@ -1130,24 +1130,29 @@ def map_train_to_library(ctx, params):
...
+        ctx.fail(
+            "Cannot combine LoRA with a padding free model. Set lora_rank to 0 or disable is_padding_free"
+        )
+    if lora_options and not lora_enabled:
```
Comment (cdoern) [inline src/instructlab/configuration.py:1154]: makes sense

**Review by cdoern — APPROVED** [review]
lgtm, thanks for cleaning this up

**Comment by cdoern** [comment]
@alinaryan  PTAL

**Review by alinaryan — APPROVED** [review]
Thanks for catching this!

**Pull request merged**